tasks/main.yml

```diff
@@ -105,79 +123,58 @@
-        status_code: [200, 404]
-      register: branch
-    - name: get commit for target branch
+        status_code: [200, 404, 422]
+      register: commit
+    - name: get commit version for your configuration
       set_fact:
-        paperlessng_commit: "{{ branch.json | json_query('commit.sha') }}"
-      when: branch.status == 200
-    - block:
-        - name: check if version is commit-or-ref
-          uri:
-            url: "https://api.github.com/repos/jonaswinkler/paperless-ng/commits/{{ paperlessng_version }}"
-            method: GET
-            status_code: [200, 404, 422]
-          register: commit
-        - name: get commit for target commit-or-ref
-          set_fact:
-            paperlessng_commit: "{{ commit.json | json_query('sha') }}"
-          when: commit.status == 200
-        - name: fail
-          fail:
-            msg: "Can not determine commit from `paperlessng_version=={{ paperlessng_version }}`!"
-          when: commit.status != 200
-      when: branch.status == 404
-  when: not(paperlessng_version | regex_search("^(ng-)?(\d+\.\d+\.\d+)$"))
-
-- name: check for paperless-ng installation
+        paperlessng_commit: "{{ commit.json | json_query('sha') }}"
+      when: commit.status == 200
+    - name: get commit for target commit-or-ref check
+      fail:
+        msg: "Can not determine commit from `paperlessng_version=={{ paperlessng_version }}`!"
+      when: commit.status != 200
+
+- name: check for existing paperless-ngx version file
+  ansible.builtin.stat:
+    path: "{{ paperlessng_directory }}/.installed_version"
+  register: paperlessng_current_commitfile
+
+- name: get the current installed version as commit
   command:
     cmd: "cat {{ paperlessng_directory }}/.installed_version"
-  changed_when: '"No such file or directory" in paperlessng_current_commit.stderr or paperlessng_current_commit.stdout != paperlessng_commit | string'
+  changed_when: 'paperlessng_current_commit.stdout != paperlessng_commit | string'
   failed_when: false
   ignore_errors: yes
   register: paperlessng_current_commit
+  when: paperlessng_current_commitfile.stat.exists
 
 - name: register current state
   set_fact:
-    fresh_installation: '{{ "No such file or directory" in paperlessng_current_commit.stderr }}'
-    update_installation: '{{ "No such file or directory" not in paperlessng_current_commit.stderr and paperlessng_current_commit.stdout != paperlessng_commit | string }}'
+    fresh_installation: '{{ not paperlessng_current_commitfile.stat.exists }}'
+    update_installation: true
+    reconfigure_only: true
+
+- name: register current existing state
+  set_fact:
+    update_installation: '{{ paperlessng_current_commit.stdout != paperlessng_commit | string }}'
     reconfigure_only: "{{ paperlessng_current_commit.stdout == paperlessng_commit | string }}"
+  when: not fresh_installation
 
 - block:
-    - name: backup current paperless-ng installation
+    - name: backup current paperless-ngx installation
```

Comment:
If `/opt/paperless-ngx` doesn't already exist, this step fails.

Reply:
The flags were wrong on the `register current state` step. I was testing the upgrade to 1.8.0. Both flags now set to follow the fresh installation path